packages/core/src/crypto/keyUtils.ts

```diff
@@ -27,7 +27,37 @@ export function isValidPrivateKey(privateKey: Buffer, keyType: KeyType): boolean
     [KeyType.P256]: 32,
     [KeyType.P384]: 48,
     [KeyType.P521]: 66,
-  }
+  } as const
 
   return Buffer.isBuffer(privateKey) && privateKey.length === privateKeyLength[keyType]
 }
+
+export function isSigningSupportedForKeyType(keyType: KeyType): boolean {
+  const keyTypeSigningSupportedMapping = {
+    [KeyType.Ed25519]: true,
```

Comment:
Do you think it wouldn't be better to define this on the Jwk class as a static method? We can keep this function but just check whether the supportedSigningAlgos has length > 1 and then we support signing. This would avoid having some duplicate code.

Reply:
Yes, I was thinking about it as well. Two limitations I see:
- This makes JWKs the source of truth, while our lowest layer is Key/KeyType
- That we don't support the alg, doesn't neccesarily mean it doesn't support signing / encrypting. So it can give incorrect results

But I do agree it would be cleaner